[Bug] [Task] ParameterUtils will throw exception when the input contains '$[xx]'

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

When the script contains $[str], and str is not a date placeholder, will throw exception, this is unexcepted, the place holder is only used to replace the parameter, if the parameter is not a time place holder, we should skip it rather than throw exception.

### What you expected to happen

-

### How to reproduce

```java
public void convertParameterPlaceholders() {
        final String sql = "get_json_object(json_array, '$[abc].pageId')";
        final Map<String, String> paramsMap = new HashMap<>();
        paramsMap.put("key", "value");

        final String result = ParameterUtils.convertParameterPlaceholders(sql, paramsMap);
        System.out.println(result);
    }
```

```java
java.lang.IllegalArgumentException: Unsupported placeholder expression: abc

	at org.apache.dolphinscheduler.plugin.task.api.parser.TimePlaceholderUtils.calculateTime(TimePlaceholderUtils.java:378)
	at org.apache.dolphinscheduler.plugin.task.api.parser.TimePlaceholderUtils.getPlaceHolderTime(TimePlaceholderUtils.java:346)
	at org.apache.dolphinscheduler.plugin.task.api.utils.ParameterUtils.dateTemplateParse(ParameterUtils.java:305)
	at org.apache.dolphinscheduler.plugin.task.api.utils.ParameterUtils.convertParameterPlaceholders(ParameterUtils.java:85)
	at org.apache.dolphinscheduler.plugin.task.api.utils.ParameterUtilsTest.convertParameterPlaceholders(ParameterUtilsTest.java:117)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
Caused by: java.lang.IllegalArgumentException: Unknown pattern letter: b
	at java.base/java.time.format.DateTimeFormatterBuilder.parsePattern(DateTimeFormatterBuilder.java:1811)
	at java.base/java.time.format.DateTimeFormatterBuilder.appendPattern(DateTimeFormatterBuilder.java:1708)
	at java.base/java.time.format.DateTimeFormatter.ofPattern(DateTimeFormatter.java:564)
	at org.apache.dolphinscheduler.common.utils.DateUtils.format(DateUtils.java:145)
	at org.apache.dolphinscheduler.common.utils.DateUtils.format(DateUtils.java:721)
	at org.apache.dolphinscheduler.plugin.task.api.parser.TimePlaceholderUtils.calculateTime(TimePlaceholderUtils.java:376)
	... 44 more
```

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parser/PropertyPlaceholderHelper.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parser/PropertyPlaceholderHelper.java
@@ -119,7 +119,7 @@ public class PropertyPlaceholderHelper {
      */
     public String replacePlaceholders(String value, PlaceholderResolver placeholderResolver) {
         notNull(value, "'value' must not be null");
-        return parseStringValue(value, placeholderResolver, new HashSet<String>());
+        return parseStringValue(value, placeholderResolver, new HashSet<>());
     }
 
     protected String parseStringValue(

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parser/TimePlaceholderUtils.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parser/TimePlaceholderUtils.java
@@ -72,44 +72,6 @@ import lombok.extern.slf4j.Slf4j;
 public class TimePlaceholderUtils {
 
     /**
-     * Prefix of the position to be replaced
-     */
-    public static final String PLACEHOLDER_PREFIX = "$[";
-
-    /**
-     * The suffix of the position to be replaced
-     */
-    public static final String PLACEHOLDER_SUFFIX = "]";
-
-    /**
-     * Replaces all placeholders of format {@code ${name}} with the value returned
-     * from the supplied {@link PropertyPlaceholderHelper.PlaceholderResolver}.
-     *
-     * @param value                          the value containing the placeholders to be replaced
-     * @param date                           custom date
-     * @param ignoreUnresolvablePlaceholders ignore unresolvable placeholders
-     * @return the supplied value with placeholders replaced inline
-     */
-    public static String replacePlaceholders(String value, Date date, boolean ignoreUnresolvablePlaceholders) {
-        PropertyPlaceholderHelper strictHelper = getPropertyPlaceholderHelper(false);
-        PropertyPlaceholderHelper nonStrictHelper = getPropertyPlaceholderHelper(true);
-
-        PropertyPlaceholderHelper helper = (ignoreUnresolvablePlaceholders ? nonStrictHelper : strictHelper);
-        return helper.replacePlaceholders(value, new TimePlaceholderResolver(value, date));
-    }
-
-    /**
-     * Creates a new {@code PropertyPlaceholderHelper} that uses the supplied prefix and suffix.
-     *
-     * @param ignoreUnresolvablePlaceholders indicates whether unresolvable placeholders should
-     *                                       be ignored ({@code true}) or cause an exception ({@code false})
-     */
-    private static PropertyPlaceholderHelper getPropertyPlaceholderHelper(boolean ignoreUnresolvablePlaceholders) {
-        return new PropertyPlaceholderHelper(PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX, null,
-                ignoreUnresolvablePlaceholders);
-    }
-
-    /**
      * calculate expression's value
      *
      * @param expression expression
@@ -303,79 +265,35 @@ public class TimePlaceholderUtils {
     }
 
     /**
-     * Placeholder replacement resolver
-     */
-    private static class TimePlaceholderResolver
-            implements
-                PropertyPlaceholderHelper.PlaceholderResolver {
-
-        private final String value;
-
-        private final Date date;
-
-        public TimePlaceholderResolver(String value, Date date) {
-            this.value = value;
-            this.date = date;
-        }
-
-        @Override
-        public String resolvePlaceholder(String placeholderName) {
-            try {
-                return calculateTime(placeholderName, date);
-            } catch (Exception ex) {
-                log.error("resolve placeholder '{}' in [ {} ]", placeholderName, value, ex);
-                return null;
-            }
-        }
-    }
-
-    /**
-     * return the formatted date according to the corresponding date format
+     * Format time expression with given date, the date cannot be null.
+     * <p> If the expression is not a time expression, return the original expression.
      *
-     * @param expression date expression
-     * @param date       date
-     * @return reformat date
      */
-    public static String getPlaceHolderTime(String expression, Date date) {
-        if (StringUtils.isBlank(expression)) {
-            throw new IllegalArgumentException("expression is null");
-        }
-        if (null == date) {
-            throw new IllegalArgumentException("date is null");
-        }
-        return calculateTime(expression, date);
-    }
-
-    /**
-     * calculate time
-     *
-     * @param date date
-     * @return calculate time
-     */
-    private static String calculateTime(String expression, Date date) {
+    public static String formatTimeExpression(final String timeExpression, final Date date,
+                                              final boolean ignoreInvalidExpression) {
         // After N years: $[add_months(yyyyMMdd,12*N)], the first N months: $[add_months(yyyyMMdd,-N)], etc
         if (date == null) {
-            throw new IllegalArgumentException("Cannot parse the expression: " + expression + ", date is null");
+            throw new IllegalArgumentException("Cannot parse the expression: " + timeExpression + ", date is null");
+        }
+        if (StringUtils.isEmpty(timeExpression)) {
+            if (ignoreInvalidExpression) {
+                return timeExpression;
+            }
+            throw new IllegalArgumentException("Cannot format the date" + date + " with null timeExpression");
         }
         try {
-            if (expression.startsWith(TIMESTAMP)) {
-                String timeExpression = expression.substring(TIMESTAMP.length() + 1, expression.length() - 1);
-
-                Map.Entry<Date, String> entry = calcTimeExpression(timeExpression, date);
-
-                String dateStr = DateUtils.format(entry.getKey(), entry.getValue());
-
-                Date timestamp = DateUtils.parse(dateStr, PARAMETER_FORMAT_TIME);
-
-                return String.valueOf(timestamp.getTime() / 1000);
+            if (timeExpression.startsWith(TIMESTAMP)) {
+                return calculateTimeStamp(timeExpression, date);
             }
-            if (expression.startsWith(YEAR_WEEK)) {
-                return calculateYearWeek(expression, date);
+            if (timeExpression.startsWith(YEAR_WEEK)) {
+                return calculateYearWeek(timeExpression, date);
             }
-            Map.Entry<Date, String> entry = calcTimeExpression(expression, date);
-            return DateUtils.format(entry.getKey(), entry.getValue());
+            return calcTimeExpression(timeExpression, date);
         } catch (Exception e) {
-            throw new IllegalArgumentException("Unsupported placeholder expression: " + expression, e);
+            if (ignoreInvalidExpression) {
+                return timeExpression;
+            }
+            throw new IllegalArgumentException("Unsupported placeholder expression: " + timeExpression, e);
         }
     }
 
@@ -385,7 +303,7 @@ public class TimePlaceholderUtils {
      * @param date       date
      * @return week of year
      */
-    public static String calculateYearWeek(String expression, Date date) {
+    private static String calculateYearWeek(final String expression, final Date date) {
 
         String dataFormat = expression.substring(YEAR_WEEK.length() + 1, expression.length() - 1);
 
@@ -465,6 +383,19 @@ public class TimePlaceholderUtils {
         return weekYearStr;
     }
 
+    private static String calculateTimeStamp(final String expression, final Date date) {
+        final String timeExpression = expression.substring(TIMESTAMP.length() + 1, expression.length() - 1);
+        final String calculatedTimeExpression = calcTimeExpression(timeExpression, date);
+        if (StringUtils.equals(expression, calculatedTimeExpression)) {
+            return expression;
+        }
+        final Date timestamp = DateUtils.parse(calculatedTimeExpression, PARAMETER_FORMAT_TIME);
+        if (timestamp == null) {
+            throw new IllegalArgumentException("Cannot parse the expression: " + expression + " with date: " + date);
+        }
+        return String.valueOf(timestamp.getTime() / 1000);
+    }
+
     /**
      * calculate time expresstion
      *
@@ -472,8 +403,8 @@ public class TimePlaceholderUtils {
      * @param date       date
      * @return map with date, date format
      */
-    public static Map.Entry<Date, String> calcTimeExpression(String expression, Date date) {
-        Map.Entry<Date, String> resultEntry;
+    private static String calcTimeExpression(final String expression, final Date date) {
+        final Map.Entry<Date, String> resultEntry;
 
         if (expression.startsWith(ADD_MONTHS)) {
             resultEntry = calcMonths(expression, date);
@@ -500,8 +431,9 @@ public class TimePlaceholderUtils {
         } else {
             resultEntry = calcMinutes(expression, date);
         }
-
-        return resultEntry;
+        final Date finalDate = resultEntry.getKey();
+        final String finalFormat = resultEntry.getValue();
+        return DateUtils.format(finalDate, finalFormat);
     }
 
     /**

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ParameterUtils.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ParameterUtils.java
@@ -49,8 +49,6 @@ public class ParameterUtils {
 
     private static final Pattern DATE_PARSE_PATTERN = Pattern.compile("\\$\\[([^\\$\\]]+)]");
 
-    private static final Pattern DATE_START_PATTERN = Pattern.compile("^[0-9]");
-
     private static final char PARAM_REPLACE_CHAR = '?';
 
     private ParameterUtils() {
@@ -288,21 +286,24 @@ public class ParameterUtils {
         return map;
     }
 
-    private static String dateTemplateParse(String templateStr, Date date) {
-        if (templateStr == null) {
-            return null;
+    // Replace the time placeholder in the template string with the given actual time value
+    // If the templateStr does not contain a time placeholder, will return the original string
+    private static String dateTemplateParse(final String templateStr, final Date date) {
+        if (StringUtils.isEmpty(templateStr)) {
+            return templateStr;
         }
 
-        StringBuffer newValue = new StringBuffer(templateStr.length());
+        final StringBuffer newValue = new StringBuffer(templateStr.length());
 
-        Matcher matcher = DATE_PARSE_PATTERN.matcher(templateStr);
+        final Matcher matcher = DATE_PARSE_PATTERN.matcher(templateStr);
 
         while (matcher.find()) {
-            String key = matcher.group(1);
-            if (DATE_START_PATTERN.matcher(key).matches()) {
+            final String key = matcher.group(1);
+            String value = TimePlaceholderUtils.formatTimeExpression(key, date, true);
+            if (StringUtils.equals(key, value)) {
+                // There is no corresponding time format, so the original string is retained
                 continue;
             }
-            String value = TimePlaceholderUtils.getPlaceHolderTime(key, date);
             matcher.appendReplacement(newValue, value);
         }
 

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/TimePlaceholderUtilsTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/TimePlaceholderUtilsTest.java
@@ -27,7 +27,6 @@ import org.apache.dolphinscheduler.plugin.task.api.utils.ParameterUtils;
 import java.util.Date;
 import java.util.Map;
 
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class TimePlaceholderUtilsTest {
@@ -131,10 +130,8 @@ public class TimePlaceholderUtilsTest {
     }
 
     @Test
-    void getPlaceHolderTime() {
-        IllegalArgumentException illegalArgumentException = Assertions.assertThrows(IllegalArgumentException.class,
-                () -> TimePlaceholderUtils.getPlaceHolderTime("$[week_last_day(yyyy-MM-dd,0) - 1]", new Date()));
-        assertEquals("Unsupported placeholder expression: $[week_last_day(yyyy-MM-dd,0) - 1]",
-                illegalArgumentException.getMessage());
+    void formatTimeExpressionWithInvalidExpression() {
+        assertEquals("$[week_last_day(yyyy-MM-dd,0) - 1]",
+                TimePlaceholderUtils.formatTimeExpression("$[week_last_day(yyyy-MM-dd,0) - 1]", new Date(), true));
     }
 }

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/utils/ParameterUtilsTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/utils/ParameterUtilsTest.java
@@ -17,18 +17,16 @@
 
 package org.apache.dolphinscheduler.plugin.task.api.utils;
 
-import static org.apache.dolphinscheduler.plugin.task.api.parser.TimePlaceholderUtils.replacePlaceholders;
+import static org.apache.dolphinscheduler.plugin.task.api.parser.PlaceholderUtils.replacePlaceholders;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import org.apache.dolphinscheduler.common.constants.DateConstants;
-import org.apache.dolphinscheduler.common.utils.DateUtils;
 import org.apache.dolphinscheduler.common.utils.JSONUtils;
 import org.apache.dolphinscheduler.plugin.task.api.enums.DataType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.Direct;
 import org.apache.dolphinscheduler.plugin.task.api.model.Property;
-import org.apache.dolphinscheduler.plugin.task.api.parser.PlaceholderUtils;
 
 import java.text.ParseException;
-import java.util.Date;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -51,8 +49,8 @@ public class ParameterUtilsTest {
                 JSONUtils.toJsonString(Lists.newArrayList(3.1415, 2.44, 3.44))));
         String sql = ParameterUtils.expandListParameter(params,
                 "select * from test where col1 in ('${col1}') and date='${date}' and col2 in ('${col2}')");
-        Assertions.assertEquals("select * from test where col1 in (?,?,?) and date=? and col2 in (?,?,?)", sql);
-        Assertions.assertEquals(7, params.size());
+        assertEquals("select * from test where col1 in (?,?,?) and date=? and col2 in (?,?,?)", sql);
+        assertEquals(7, params.size());
 
         Map<Integer, Property> params2 = new HashMap<>();
         params2.put(1,
@@ -60,8 +58,8 @@ public class ParameterUtilsTest {
         params2.put(2, new Property("date", Direct.IN, DataType.DATE, "2020-06-30"));
         String sql2 = ParameterUtils.expandListParameter(params2,
                 "select * from test where col1 in ('${col}') and date='${date}'");
-        Assertions.assertEquals("select * from test where col1 in (?) and date=?", sql2);
-        Assertions.assertEquals(2, params2.size());
+        assertEquals("select * from test where col1 in (?) and date=?", sql2);
+        assertEquals(2, params2.size());
 
     }
 
@@ -77,21 +75,14 @@ public class ParameterUtilsTest {
 
         // parameterString、parameterMap is not null
         String parameterString = "test_parameter";
-        Assertions.assertEquals(parameterString,
+        assertEquals(parameterString,
                 ParameterUtils.convertParameterPlaceholders(parameterString, parameterMap));
 
         // replace variable ${} form
         parameterMap.put("testParameter2", "${testParameter}");
-        Assertions.assertEquals(parameterString,
-                PlaceholderUtils.replacePlaceholders(parameterString, parameterMap, true));
+        assertEquals(parameterString,
+                replacePlaceholders(parameterString, parameterMap, true));
 
-        // replace time $[...] form, eg. $[yyyyMMdd]
-        Date cronTime = new Date();
-        Assertions.assertEquals(parameterString, replacePlaceholders(parameterString, cronTime, true));
-
-        // replace time $[...] form, eg. $[yyyyMMdd]
-        Date cronTimeStr = DateUtils.stringToDate("2019-02-02 00:00:00");
-        Assertions.assertEquals(parameterString, replacePlaceholders(parameterString, cronTimeStr, true));
     }
 
     @Test
@@ -103,20 +94,30 @@ public class ParameterUtilsTest {
         parameterMap.put("user", "Kris");
         parameterMap.put(DateConstants.PARAMETER_DATETIME, "20201201123000");
         parameterString = ParameterUtils.convertParameterPlaceholders(parameterString, parameterMap);
-        Assertions.assertEquals(
+        assertEquals(
                 "Kris is userName, '$[1]' '20221201' '20181201' '20210301' '20200801' '20201215' '20201117'  '20201204'  '$[0]' '20201128' '143000' '113000' '123300' '122800'  '$[3]'",
                 parameterString);
+    }
+
+    @Test
+    public void convertParameterPlaceholdersWithPunct() {
+        final String sql = "get_json_object(json_array, '$[*].pageId')";
+        final Map<String, String> paramsMap = new HashMap<>();
+        paramsMap.put("key", "value");
+
+        assertEquals("get_json_object(json_array, '$[*].pageId')",
+                ParameterUtils.convertParameterPlaceholders(sql, paramsMap));
     }
 
     /**
      * Test handleEscapes
      */
     @Test
-    public void testHandleEscapes() throws Exception {
+    public void testHandleEscapes() {
         Assertions.assertNull(ParameterUtils.handleEscapes(null));
-        Assertions.assertEquals("", ParameterUtils.handleEscapes(""));
-        Assertions.assertEquals("test Parameter", ParameterUtils.handleEscapes("test Parameter"));
-        Assertions.assertEquals("////%test////%Parameter", ParameterUtils.handleEscapes("%test%Parameter"));
+        assertEquals("", ParameterUtils.handleEscapes(""));
+        assertEquals("test Parameter", ParameterUtils.handleEscapes("test Parameter"));
+        assertEquals("////%test////%Parameter", ParameterUtils.handleEscapes("%test%Parameter"));
     }
 
 }

